napari/layers/shapes/_shape_list.py

```diff
@@ -1113,17 +1121,26 @@ def inside(self, coord):
             Index of shape if any that is at the coordinates. Returns `None`
             if no shape is found.
         """
-        triangles = self._mesh.vertices[self._mesh.displayed_triangles]
-        indices = inside_triangles(triangles - coord)
-        shapes = self._mesh.displayed_triangles_index[indices, 0]
-
-        if len(shapes) == 0:
+        if not self.shapes:
+            return None
+        bounding_boxes = self._bounding_boxes
+        inside = np.all(
+            (bounding_boxes[0] <= coord) * (bounding_boxes[1] >= coord),
+            axis=1,
+        )
+        inside_indices = np.nonzero(inside)[0]
+        if not inside_indices.size:
```

Comment:
For naturally integer values I think it is more readable to compare rather than to check for truthiness.

```suggestion
        if inside_indices.size == 0:
```